source/magpie/group.d

```diff
@@ -463,6 +463,146 @@ public:
+        * can be avoided by rewriting the implementation above to account for everything.
+        */
+
+        int[2] pos;
+        int p = getGroupPosition(groupTitle);
+        assert(p > -1, "Group index out of bound");
+        pos[0] = p;
+
+        if(axis)
+        {
+            p = positionInGroup!(1)(index, pos[0]);
+            assert(p > -1, "Index out of bound");
+            pos[1] = p;
+        }
+        else
+        {
+            p = positionInGroup!(0)(index, pos[0]);
+            assert(p > -1, "Index out of bound");
+            pos[1] = p;
+        }
+
+        static if(is(T[0] == void))
+        {
+            assert(elements.data.length == elementCountTill[pos[0] + 1] - elementCountTill[pos[0]],
+                "Size of data doesn't match size of group column");
+            static foreach(i; 0.. GrpRowType.length)
+            {
+                if(i == pos[1])
+                {
+                    foreach(j; elementCountTill[pos[0]] .. elementCountTill[pos[0] + 1])
+                    {
+                        import std.variant: VariantException;
+                        try
+                        {
+                            static if(op == "+")
+                                data[i][j] += elements.data[j - elementCountTill[pos[0]]].get!(GrpRowType[i]);
+                            else static if(op == "-")
+                                data[i][j] -= elements.data[j - elementCountTill[pos[0]]].get!(GrpRowType[i]);
+                            else static if(op == "*")
+                                data[i][j] *= elements.data[j - elementCountTill[pos[0]]].get!(GrpRowType[i]);
+                            else static if(op == "/")
+                                data[i][j] /= elements.data[j - elementCountTill[pos[0]]].get!(GrpRowType[i]);
+                        }
+                        catch(VariantException e)
+                        {
+                            static if(op == "+")
+                                data[i][j] += cast(GrpRowType[i])elements.data[j - elementCountTill[pos[0]]].get!(double);
+                            else static if(op == "-")
+                                data[i][j] -= cast(GrpRowType[i])elements.data[j - elementCountTill[pos[0]]].get!(double);
+                            else static if(op == "*")
+                                data[i][j] *= cast(GrpRowType[i])elements.data[j - elementCountTill[pos[0]]].get!(double);
+                            else static if(op == "/")
+                                data[i][j] /= cast(GrpRowType[i])elements.data[j - elementCountTill[pos[0]]].get!(double);
+                        }
+                    }
+                }
+            }
+        }
+        else
+        {
+            assert(elements.data.length == GrpRowType.length, "Size of data doesn't match size of group column");
+            static foreach(i; 0.. GrpRowType.length)
+            {
+                static foreach(i; 0 .. GrpRowType.length)
```

Comment:
Is one of these supposed to be `Col`? also the variables `i` will collide.

Reply:
The inner loop was't necessary. My bad! 
Addressed in 6517b4a